### PR TITLE
Clickable Demos Feature UI

### PR DESCRIPTION
**Pull request by MSamiTariq**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

Refers: https://github.com/RobinNagpal/dodao-ui/issues/214

### Code Review
- [ ] I have carefully reviewed my code and believe it is final from my side.
- [ ] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [ ] I have tested the new feature or updates in both light and dark themes.
- [ ] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [ ] I have tested my changes on mobile screen sizes.
- [ ] I have tested my changes on tablet screen sizes.
- [ ] I have tested my changes on laptop screen sizes.

### Functional Testing
- [ ] I have thoroughly tested the functionality of my changes.
- [ ] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

### CODE REVIEW

**Comment by MSamiTariq** [comment]
For styling of the tooltip, we need to add the following code to the downloaded html file:

```
<script src="https://unpkg.com/@popperjs/core@2"></script> <!-- Popper.js for Tippy.js -->
<script src="https://unpkg.com/tippy.js@6"></script>
<link rel="stylesheet" href="https://unpkg.com/tippy.js@6/dist/backdrop.css" />
<link rel="stylesheet" href="https://unpkg.com/tippy.js@6/themes/material.css" />
<link rel="stylesheet" href="https://unpkg.com/tippy.js@6/animations/shift-toward.css" />
<style>
    .tippy-box[data-theme~='material'] {
        border-radius: 0.7rem;
        border: 1px solid rgb(224, 224, 224);
        box-shadow: inset 0 0 4px rgba(0, 0, 0, 0.2);
        background-color: #212429;
    }

    .tippy-box[data-theme~="material"][data-placement^="bottom"]>.tippy-arrow::before {
        border-bottom-color: #212429;
    }

    .tippy-box[data-theme~="material"][data-placement^="bottom"]>.tippy-arrow::after {
        content: "";
        position: absolute;
        border-color: transparent;
        border-style: solid;
        border-width: 9px;
        border-bottom-color: rgb(224, 224, 224);
        z-index: -1;
        top: -18px;
        left: -1px;
        width: 100%;
    }

    .tippy-box[data-theme~="material"][data-placement^="top"]>.tippy-arrow::before {
        border-top-color: #212429;
    }

    .tippy-box[data-theme~="material"][data-placement^="top"]>.tippy-arrow::after {
        content: "";
        position: absolute;
        border-color: transparent;
        border-style: solid;
        border-width: 9px;
        border-bottom-color: rgb(224, 224, 224);
        z-index: -1;
        top: 16px;
        left: -1px;
        width: 100%;
        rotate: 180deg;
    }

    /* Responsive adjustments */
    @media (max-width: 600px) {
        .tippy-box[data-theme~='material'] {
            min-width: 250px;
            /* Smaller width */
            font-size: 0.9rem;
            /* Smaller text */
        }
    }

    @media (min-width: 601px) and (max-width: 1024px) {
        .tippy-box[data-theme~='material'] {
            min-width: 300px;
            /* Medium width */
            font-size: 1rem;
            /* Medium text */
        }
    }

    @media (min-width: 1025px) {
        .tippy-box[data-theme~='material'] {
            min-width: 350px;
            /* Larger width */
            font-size: 1.1rem;
            /* Larger text */
        }
    }
</style>
```

This code I usually insert after the first meta tag

**Comment by MSamiTariq** [comment]
We need to add the following code to each of the downloaded HTML file:

```
<style>
 .tippy-box[data-theme~='material'] {
    border-radius: 0.7rem;
    border: 1px solid rgb(224, 224, 224);
    box-shadow: inset 0 0 4px rgba(0, 0, 0, 0.2);
    background-color: #212429;
}
.tippy-box[data-theme~="material"][data-placement^="bottom"] > .tippy-arrow::before {
    border-bottom-color: #212429;
}
.tippy-box[data-theme~="material"][data-placement^="bottom"] > .tippy-arrow::after {
    content: "";
    position: absolute;
    border-color: transparent;
    border-style: solid;
    border-width: 9px;
    border-bottom-color: rgb(224, 224, 224);
    z-index: -1;
    top: -18px;
    left: -1px;
}
.tippy-box[data-theme~="material"][data-placement^="top"] > .tippy-arrow::before {
    border-top-color: #212429;
}

.tippy-box[data-theme~="material"][data-placement^="top"] > .tippy-arrow::after {
    content: "";
    position: absolute;
    border-color: transparent;
    border-style: solid;
    border-width: 9px;
    border-bottom-color: rgb(224, 224, 224);
    z-index: -1;
    top: 16px;
    left: -1px;
    rotate: 180deg;
}

</style>

<script src="https://unpkg.com/@popperjs/core@2"></script>
    <script src="https://unpkg.com/tippy.js@6"></script>
    <link rel="stylesheet" href="https://unpkg.com/tippy.js@6/animations/shift-toward.css" />
    <link rel="stylesheet" href="https://unpkg.com/tippy.js@6/themes/material.css" />
    <script defer>
      // With the above scripts loaded, you can call `tippy()` with a CSS
      // selector and a `content` prop:
      window.addEventListener('message', (event) => {

        const xpathResult = document.evaluate(
        event.data.elementXPath,
        document,
        null,
        XPathResult.FIRST_ORDERED_NODE_TYPE,
        null
      );
      const target = xpathResult.singleNodeValue;

      const tooltipContent = document.createElement("div");

      // Add text content or any other elements to your tooltip as needed
      const textElement = document.createElement("p");
      textElement.textContent = event.data.tooltipContent;
      textElement.style.display = "inline";
      textElement.style.margin = "0 auto";
      textElement.style.fontSize = "1rem";
      textElement.style.fontFamily = "sans-serif";
      textElement.style.fontWeight = "300";

      tooltipContent.style.display = "flex";
      tooltipContent.style.flexDirection = "column";
      tooltipContent.style.justifyContent = "space-around";
      tooltipContent.style.minHeight = "130px"; // Set a minimum height for the tooltip
      tooltipContent.style.minWidth = "300px"; // Set a minimum width for the tooltip
      tooltipContent.style.padding = "3px 12px"; // Add padding to the tooltip
      tooltipContent.appendChild(textElement);

      // textElement.style.marginBottom = "8px"; // Add some space below the text content

      // Add an <hr> element to serve as a horizontal line
      const horizontalLine = document.createElement("hr");
      horizontalLine.style.borderTop = "1px solid #808080"; // Style the line as needed
      horizontalLine.style.margin = "2px 0"; // Add some space around the line
      // Append the horizontal line before the buttons
      tooltipContent.appendChild(horizontalLine);

      const buttonsRow = document.createElement("div");
      buttonsRow.style.display = "flex";
      buttonsRow.style.justifyContent = "space-between";

      // Create the 'Back' button
      const backButton = document.createElement("button");
      backButton.textContent = "Back";
      backButton.onclick = () => {
          event.source.postMessage(
          { backButton: true},
          event.origin,
        );
      };

      // Style the 'Back' button
      backButton.style.padding = "5px 10px"; // Padding for a larger click area
      backButton.style.border = "none"; // Remove default border
      backButton.style.borderRadius = "5px"; // Rounded corners
      backButton.style.backgroundImage =
        "linear-gradient(to right, #6e85b7, #b8c0ff)"; // Gradient background
      backButton.style.color = "white"; // Text color
      backButton.style.fontWeight = "bold"; // Make the text bold
      backButton.style.cursor = "pointer"; // Cursor on hover
      backButton.style.boxShadow = "0 2px 4px rgba(0,0,0,0.2)"; // Box shadow for depth
      backButton.style.transition = "all 0.3s ease"; // Transition for smooth hover effect

      // Align the button at the bottom left
      backButton.style.marginTop = "auto";
      backButton.style.alignSelf = "flex-start";
      backButton.style.marginRight = "auto";

      // Hover effect
      backButton.onmouseover = () => {
        backButton.style.backgroundImage =
          "linear-gradient(to right, #b8c0ff, #6e85b7)";
      };
      backButton.onmouseout = () => {
        backButton.style.backgroundImage =
          "linear-gradient(to right, #6e85b7, #b8c0ff)";
      };
      if (event.data.currentTooltipIndex === 0) backButton.style.visibility = "hidden";

      // Append the button to the container
      buttonsRow.appendChild(backButton);

      const indices = document.createElement("span");
      indices.textContent = `${event.data.currentTooltipIndex + 1} of ${
        event.data.tooltipArrayLen
      }`;

      indices.style.color = "#84868a";
      indices.style.margin = "auto";
      indices.style.fontSize = "small";
      indices.style.fontWeight = "300";

      buttonsRow.appendChild(indices);

      // Create the 'Next' button
      const nextButton = document.createElement("button");
      nextButton.textContent =
        event.data.currentTooltipIndex === event.data.tooltipArrayLen - 1 ? "Complete" : "Next";
      nextButton.onclick = async () => {
          event.source.postMessage(
          { nextButton: true},
          event.origin,
        );
      };

      // Style the 'Next' button
      nextButton.style.padding = "5px 10px"; // Padding for a larger click area
      nextButton.style.border = "none"; // Remove default border
      nextButton.style.borderRadius = "5px"; // Rounded corners
      nextButton.style.backgroundImage =
        "linear-gradient(to left, #FFB6C1, #FF69B4)"; // Gradient background, pink hues
      nextButton.style.color = "white"; // Text color
      nextButton.style.fontWeight = "bold"; // Make the text bold
      nextButton.style.cursor = "pointer"; // Cursor on hover
      nextButton.style.boxShadow = "0 2px 4px rgba(0,0,0,0.2)"; // Box shadow for depth
      nextButton.style.transition = "all 0.3s ease"; // Transition for smooth hover effect

      // Align the button at the bottom right
      nextButton.style.marginTop = "auto";
      nextButton.style.alignSelf = "flex-end";
      nextButton.style.marginLeft = "auto";
      nextButton.addEventListener('click', () => {
        console.log("next button clicked");
      });

      // Hover effect
      nextButton.onmouseover = () => {
        nextButton.style.backgroundImage =
          "linear-gradient(to left, #FF69B4, #FFB6C1)";
      };
      nextButton.onmouseout = () => {
        nextButton.style.backgroundImage =
          "linear-gradient(to left, #FFB6C1, #FF69B4)";
      };

      if (event.data.currentTooltipIndex === event.data.tooltipArrayLen - 1)
        nextButton.disabled = true;

      // Append the button to the container
      buttonsRow.appendChild(nextButton);

      tooltipContent.appendChild(buttonsRow);

      if(target){
        target.scrollIntoView({
          behavior: "smooth", // Optional: defines the transition animation
          block: "center", // Vertical alignment: options are 'start', 'center', 'end', or 'nearest'
          inline: "nearest", // Horizontal alignment: options are 'start', 'center', 'end', or 'nearest'
        });
      }

       tippy(target, {
        allowHTML: true,
        placement: "top",
        offset: [0, 10],
        animation: "shift-toward", // Use the 'scale' animation
        interactive: true,
        inertia: true,
        duration: [2000, 250],
        delay: [500, 200],
        content: tooltipContent,
        showOnCreate: true,
        hideOnClick: false,
        trigger: "manual",
        theme: "material",
      });
});
</script>
```

**Comment by MSamiTariq** [comment]
Due to issue with file handling on client side, I have uploaded the HTML files to S3 bucket and pasted the URL in `tooltipArray` for testing purposes. 

Before merging this PR, make sure to remove the hardcoded `tooltipArray` and line number 27 in `ClickableDemoModal.tsx` and also uncomment line number 28 in the same file.

File: src/app/clickable-demos/edit/[[...demoId]]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import { useI18 } from '@/hooks/useI18';
+import SingleCardLayout from '@/layouts/SingleCardLayout';
+import { ClickableDemoErrors } from '@/types/errors/clickableDemoErrors';
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+import { useEffect } from 'react';
+import styled from 'styled-components';
+
+const Wrapper = styled.div`
+  .wrapper {
```
Comment (RobinNagpal) [inline src/app/clickable-demos/edit/[[...demoId]]/page.tsx:21]: Convert to SCSS

File: src/app/clickable-demos/edit/[[...demoId]]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                    error={inputError('excerpt') ? 'Excerpt is required' : ''}
+                    maxLength={32}
+                    onUpdate={(newValue) => updateClickableDemoFunctions.updateClickableDemoField('excerpt', newValue)}
+                  >
+                    <div>Excerpt*</div>
+                  </Input>
+                </div>
+              </Block>
+
+              {clickableDemo ? (
```
Comment (RobinNagpal) [inline src/app/clickable-demos/edit/[[...demoId]]/page.tsx:111]: Seems like this condition needs to be updated

File: src/components/clickableDemos/Edit/ClickableDemoStepperItem.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  stepIndex: number;
+  stepErrors: ClickableDemoStepError | undefined;
+  moveStepUp?: (stepUuid: string) => void;
+  moveStepDown?: (stepUuid: string) => void;
+  removeStep?: (stepUuid: string) => void;
+  onUpdateStep: (step: ClickableDemoStepInput) => void;
+}
+
+const StyledStepContainer = styled.div<{ error: boolean }>`
+  border: ${(props) => (props.error ? '1px solid red' : 'none')};
```
Comment (RobinNagpal) [inline src/components/clickableDemos/Edit/ClickableDemoStepperItem.tsx:25]: Remove styled components

File: src/components/clickableDemos/List/ClickableDemoAdminDropDown.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+interface ClickableDemoAdminDropdownProps {
+  clickableDemo: ClickableDemo;
+}
+export default function ClickableDemoAdminDropdown({ clickableDemo }: ClickableDemoAdminDropdownProps) {
+  const router = useRouter();
+  const getThreeDotItems = [{ label: 'Edit', key: 'edit' }];
+
+  return (
+    <>
```
Comment (RobinNagpal) [inline src/components/clickableDemos/List/ClickableDemoAdminDropDown.tsx:15]: Empty fragment is not needed

File: src/components/clickableDemos/View/ClickableDemoModal.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+import { ClickableDemoStep, ClickableDemoWithSteps } from '@/graphql/generated/generated-types';
+import React, { useEffect, useState } from 'react';
+
+interface ClickableDemoModalProps {
+  clickableDemoWithSteps: ClickableDemoWithSteps;
+  onClose: () => void;
+}
+
+function ClickableDemoModal({ clickableDemoWithSteps, onClose }: ClickableDemoModalProps) {
+  const tooltipArray = [
```
Comment (RobinNagpal) [inline src/components/clickableDemos/View/ClickableDemoModal.tsx:11]: This should be removed
